### DISCUSSION TITLE
The `eglot-inlay-hints-mode` thingy does not toggle

### DISCUSSION THREAD

**Discussion by refaelsh** [joaotavora/eglot#1245] [General]
The docs state this:
```
M-x eglot-inlay-hints-mode

    This command toggles LSP inlay hints on and off for the current buffer. Inlay hints are small text annotations to specific parts of the whole buffer, not unlike diagnostics, but designed to help readability instead of indicating problems. For example, a C++ language server can serve hints about positional parameter names in function calls and a variable’s automatically deduced type. Inlay hints help the user not have to remember these things by heart. 
```

1. For me, it does not toggle. It always says the same thing:
```
Eglot-Inlay-Hints mode disabled in current buffer
```
2. Maybe the doc should state the default of this? Is this mode on or off?
3. I tried it in a Haskell buffer. Maybe for Haskell this mode is irrelevant?

**Comment by manuel-uberti** [comment]
In order to understand the actual problem, could you try following the instructions at https://joaotavora.github.io/eglot/#Troubleshooting-Eglot?

**Reply by refaelsh** [reply]
Sure. Gladly. I will go over this part of the doc and get back to you here.

**Reply by refaelsh** [reply]
I when over the link and it did not help. Please advice :-)

**Reply by joaotavora** [reply]
I think what @manuel-uberti is trying to say is that we can't reproduce your problem.  At least I can't.  When I `M-x eglot-inlay-hints-mode` does toggle on and off.  So you must have some odd condition in your configuration or a indeed a bug in Eglto and the only way to find it out is to report a tightly controlled experiment with the least amount of unknown variables.  And the process to do so is described in the link Manuel posted.

**Reply by refaelsh** [reply]
> When I M-x eglot-inlay-hints-mode does toggle on and off

For what language it is working for you?

> tightly controlled experiment with the least amount of unknown variables. And the process to do so is described in the link Manuel posted.

I dont see such a process described in the link. I read it again just now.

**Reply by refaelsh** [reply]
Also, is it possible that Eglot is behaving this way because Haskell's LSP does not support inlay hints (I checked, it does not)?

**Reply by refaelsh** [reply]
I just tried in Rust and everything works perfectly. May this is true:
>  Also, is it possible that Eglot is behaving this way because Haskell's LSP does not support inlay hints (I checked, it does not)?

**Reply by joaotavora** [reply]
Yes, that's probably it

**Reply by refaelsh** [reply]
> Yes, that's probably it

Then, may I please suggest adding this fact to the docs? The fact that if the LSP does not support inlay hints, it would not toggle.
If that's ok by you, I can open a pull request for it.
